src/SelectSingle.vue

```diff
@@ -64,7 +66,13 @@
 		},
 		data(): ComponentData {
 			const activeIndex = this.value
-				? this.options.findIndex(currentOption => currentOption.value == this.value.value)
+				? this.options.findIndex(currentOption => {
+                    const { value: currentValue } = this.value
+                    if (Array.isArray(currentOption.value) && Array.isArray(currentValue)) {
+                        return currentOption.value.join(arrayJoinDelimiter) === currentValue.join(arrayJoinDelimiter)
+                    }
```

Comment:
I would prefer to leave this logic about `join` in consuming projects and leave `activeIndex` as it was before
```
const activeIndex = this.value
				? this.options.findIndex(currentOption => currentOption.value == this.value.value)
				: 0
```

Reply:
Ivan is exactly right here and everyone should listen to him at all times :) He helped walk through why this code does not need to change in a very gentle and thoughtful way